### PR TITLE
config-tls-generic: Add config option for MBEDTLS_PKCS1_V21

### PR DESCRIPTION
**Pull request by eugenecohen**
PKCS1 v2.1 defines a more secure scheme by making use of
RSAES-OAEP and RSAASSA-PSS.  Add a config flag to enable
this capability.

Signed-off-by: Eugene Cohen <eugene@nuviainc.com>

### CODE REVIEW

**Comment by eugenecohen** [comment]
@carlescufi @rlubos @ceolin

File: configs/config-tls-generic.h [hunk 1]
```diff
@@ -297,6 +297,12 @@
 #define MBEDTLS_CIPHER_C
 #endif
 
+#if defined(CONFIG_MBEDTLS_PKCS1_V21_ENABLED)
+#define MBEDTLS_PKCS1_V21
+#endif
+
+/* Automatic dependencies */
```
Comment (rlubos) [inline configs/config-tls-generic.h:304]: nit: Duplicated comment.

Comment (eugenecohen) [reply configs/config-tls-generic.h]: whoops, that came in with my merge from master, sorry about that, fixed

**Comment by eugenecohen** [comment]
@d3zd3z it looks like the config-tls-generic file is already unique to the zephyr fork so this change is not independently upstreamable.  It would be great if the whole file could be upstreamed to minimize differences, not sure if the mbedtls project would accept it.

Thanks!

**Comment by eugenecohen** [comment]
what are the next steps on this PR - is there something on my side i need to do?  thanks.

**Comment by rlubos** [comment]
We need someone with a merge right to show up. @carlescufi @nashif

**Pull request merged**